### ISSUE TITLE
Attachments from URL not follow redirection

### ISSUE THREAD

**Issue by rapee** [andris9/Nodemailer#488]
When I try to attach and embed images by URL as source, the image often get corrupted. I found out that this is because some of the URLs return 301/302 redirection. I can get the final redirect URL by other means but I think nodemailer should handle this by default.

This is the options I used:

```json
  {
    "attachments": [  {
        "filename": "item_image.png",
        "path": "https://www.mydomain.com/item/a123b456/image",
        "cid": "item_image.png"
      } ]
  }
```

Thanks for great software btw.

**Comment by andris9** [comment]
This behavior is caused by the *hyperquest* module that is heavily opinionated and among other things does not support redirects. I've been planning to replace it with something else but haven't got the time for it so far. I wouldn't use *request* because of the baggage it comes with (huge amount of dependencies) and probably would go with [wreck](https://www.npmjs.com/package/wreck).

**Comment by rapee** [comment]
Sound great! Looking forward to it :)

**Comment by andris9** [comment]
Fixed in v1.7.0 where needle is used instead of hyperquest

**Comment by rapee** [comment]
Awesome!
